programs/marinade-finance/src/instructions/admin/emergency_pause.rs

```diff
@@ -0,0 +1,47 @@
+use anchor_lang::{prelude::*};
+
+use crate::{
+    error::MarinadeError,
+    events::{admin::EmergencyPauseEvent},
+    State,
+};
+
+#[derive(Accounts)]
+pub struct EmergencyPause<'info> {
+    #[account(
+        mut,
+        has_one = pause_authority @ MarinadeError::InvalidPauseAuthority
+    )]
+    pub state: Account<'info, State>,
+    pub pause_authority: Signer<'info>,
+}
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, AnchorSerialize, AnchorDeserialize)]
+pub struct EmergencyPauseData {
+    pub set_pause: bool
```

Comment:
Please find a better name for the field. It must be not a verb. And do not use struct here. It is not required. Just use bool parameter or better make 2 separate commands pause and resume